src/extensions/exceptions/exception-autocapture.ts

```diff
@@ -0,0 +1,131 @@
+
+            window.onunhandledrejection = function (
+                this: ExceptionObserver,
+                ...args: [ev: PromiseRejectionEvent]
+            ): boolean {
+                const errorProperties: ErrorProperties = unhandledRejectionToProperties(args)
+                this.sendExceptionEvent(errorProperties)
+
+                if (this.originalOnUnhandledRejectionHandler) {
+                    // eslint-disable-next-line prefer-rest-params
+                    return this.originalOnUnhandledRejectionHandler.apply(window, args)
+                }
+
+                return true
+            }.bind(this)
+            ;(window.onunhandledrejection as any).__POSTHOG_INSTRUMENTED__ = true
+        } catch (e) {
+            console.error('PostHog failed to start exception autocapture', e)
+            this.stopCapturing()
+        }
+    }
+
+    stopCapturing() {
+        if (this.originalOnErrorHandler !== undefined) {
+            window.onerror = this.originalOnErrorHandler
+            this.originalOnErrorHandler = null
+        }
+        delete (window.onerror as any)?.__POSTHOG_INSTRUMENTED__
+
+        if (this.originalOnUnhandledRejectionHandler !== undefined) {
+            window.onunhandledrejection = this.originalOnUnhandledRejectionHandler
+            this.originalOnUnhandledRejectionHandler = null
+        }
+        delete (window.onunhandledrejection as any)?.__POSTHOG_INSTRUMENTED__
+    }
+
+    isCapturing() {
+        return !!(window.onerror as any)?.__POSTHOG_INSTRUMENTED__
+    }
+
+    isEnabled() {
+        return this.instance.get_config('exception_autocapture') ?? this.remoteEnabled ?? false
+    }
+
+    afterDecideResponse(response: DecideResponse) {
+        this.remoteEnabled = response.autocaptureExceptions || false
+        if (this.isEnabled()) {
+            this.startCapturing()
+        }
+    }
+
+    /**
+     * :TRICKY: Make sure we batch these requests
+     */
+    sendExceptionEvent(properties: { [key: string]: any }) {
+        this.instance.capture('$exception', properties, {
+            transport: 'XHR',
+            method: 'POST',
+            endpoint: EXCEPTION_INGESTION_ENDPOINT,
+            _noTruncate: true,
+            _batchKey: 'exceptionEvent',
+        })
+    }
+
+    captureException(args: ErrorEventArgs, properties?: Properties) {
+        if (this.isCapturing()) {
+            const errorProperties = errorToProperties(args)
+            const propertiesToSend = { ...properties, ...errorProperties }
+
+            const posthogHost = this.instance.config.ui_host || this.instance.config.api_host
+            if (this.instance.sessionRecordingStarted()) {
+                errorProperties.$exception_sessionRecordingURL =
+                    posthogHost +
+                    '/recordings/' +
+                    this.instance.sessionManager.checkAndGetSessionAndWindowId(true).sessionId
+            }
```

Comment:
We have `get_session_replay_url()` now which should be used here. 

One thought though - why do we want this 🤔 is it not redundant given that we can rebuild it on the other end?